### PR TITLE
Tram signal process/destination improvements

### PR DESCRIPTION
**Pull request by lessthnthree**
## About The Pull Request

- Tram crossing signals now call process() much less often
- Tram crossing signals don't turn amber/red needlessly
- Tram destination landmarks are more generic to accommodate future maps, like Birdshot
- Renames to_where and from_where, because those vars didn't always match tram position to/from

## Why It's Good For The Game

Tram works better, uses less processing

## Changelog

:cl: LT3
code: Tram crossing signal/platform logic improvements
/:cl:

### CODE REVIEW

File: code/modules/mapfluff/ruins/spaceruin_code/hilbertshotel.dm [hunk 1]
```diff
@@ -500,19 +500,22 @@ GLOBAL_VAR_INIT(hhMysteryRoomNumber, rand(1, 999999))
 /obj/effect/landmark/lift_id/hilbert
 	specific_lift_id = HILBERT_TRAM
 
-/obj/effect/landmark/tram/left_part/hilbert
+/obj/effect/landmark/tram/hilbert/left
+	name = "Port"
 	specific_lift_id = HILBERT_TRAM
-	destination_id = "left_part_hilbert"
+	platform_code = 11
```
Comment (san7890) [inline code/modules/mapfluff/ruins/spaceruin_code/hilbertshotel.dm:506]: I don't like funny numbers like these, could you make these codes defines like

```dm
#define HILBERT_LEFT 11
```

Also, why does it have to be 11

Comment (lessthnthree) [reply code/modules/mapfluff/ruins/spaceruin_code/hilbertshotel.dm]: Yeah, made em defines. It doesn't have to be 11, it was a remnant of before I knew if we could reuse codes (we can)

Switched them to 1/2/3.

File: code/modules/industrial_lift/tram/tram_machinery.dm [hunk 2]
```diff
@@ -449,18 +468,11 @@ GLOBAL_LIST_EMPTY(tram_doors)
 		tram_velocity_sign = tram.travel_direction & EAST ? 1 : -1
 
 	// How far away are we? negative if already passed.
-	var/approach_distance = tram_velocity_sign * (signal_pos - tram_pos)
+	var/approach_distance = tram_velocity_sign * (signal_pos - (tram_pos + 5))
```
Comment (Kylerace) [inline code/modules/industrial_lift/tram/tram_machinery.dm:471]: magic number, use a define

Comment (lessthnthree) [reply code/modules/industrial_lift/tram/tram_machinery.dm:471]: Got it

**Pull request merged**